measure/src/measure.rs

```diff
@@ -102,4 +133,70 @@ mod tests {
         let measure = Measure::start("test_not_stopped");
         assert_eq!(format!("{}", measure), "test_not_stopped running");
     }
+
+    fn my_multiply(x: i32, y: i32) -> i32 {
+        x * y
+    }
+
+    struct SomeStruct {
+        x: i32,
+    }
+    impl SomeStruct {
+        fn add_to(&self, x: i32) -> i32 {
+            x + self.x
+        }
+    }
+
+    #[test]
+    fn test_measure_with() {
+        // Ensure that the measurement side actually works
+        {
+            let (_result, measure) = Measure::wrap(|s| sleep(Duration::from_secs(s)), 1, "test");
+            assert!(measure.as_s() >= 0.99f32 && measure.as_s() <= 1.01f32);
+            assert!(measure.as_ms() >= 990 && measure.as_ms() <= 1_010);
+            assert!(measure.as_us() >= 999_000 && measure.as_us() <= 1_010_000);
+        }
+
+        // Ensure that wrap() can be called with a simple closure
+        {
+            let expected = 1;
+            let (actual, _measure) = Measure::wrap(|x| x, expected, "test");
+            assert_eq!(actual, expected);
+        }
+
+        // Ensure that wrap() can be called with a tuple
+        {
+            let (result, _measure) = Measure::wrap(|(x, y)| x + y, (1, 2), "test");
+            assert_eq!(result, 1 + 2);
+        }
+
+        // Ensure that wrap() can be called with a normal function
+        {
+            let (result, _measure) = Measure::wrap(|(x, y)| my_multiply(x, y), (3, 4), "test");
```

Comment:
Unfortunately no. Since `multiply()` takes two arguments, and `wrap()` only passes along one, the closure is necessary. _However_, if you make a wrapper for `multiply()` that takes a single tuple arg and then do the unwrapping and call `multiply()`, that does work.

I was really hoping that Rust had something similar to C++'s variadic template parameters, but from what I can tell, it doesn't.